hardware/detect.py

```diff
@@ -726,6 +687,86 @@ def find_element(xml, xml_spec, sys_subtype,
+    lscpux = {}
+    if lscpux_output is None:
+        output = detect_utils.output_lines('LANG=en_US.UTF-8 lscpu -x')
+    else:
+        output = lscpux_output.splitlines()
+    for line in output:
+        item, value = line.split(':')
+        lscpux[item.strip(':')] = value.strip()
+
+    hw_lst.append(("cpu", "physical", "number", int(lscpu["Socket(s)"])))
+    for processor in range(int(lscpu["Socket(s)"])):
+        hw_lst.append(('cpu', "physical_{}".format(
+            processor), 'vendor', lscpu['Vendor ID']))
+        hw_lst.append(('cpu', "physical_{}".format(
+            processor), 'product', lscpu['Model name']))
+        hw_lst.append(('cpu', "physical_{}".format(
+            processor), 'cores', int(lscpu['Core(s) per socket'])))
+        hw_lst.append(('cpu', "physical_{}".format(
+            processor), 'threads', int(lscpu['Thread(s) per core']) * int(lscpu['Core(s) per socket'])))
+        hw_lst.append(('cpu', "physical_{}".format(
+            processor), 'family', int(lscpu['CPU family'])))
+        hw_lst.append(('cpu', "physical_{}".format(
+            processor), 'model', int(lscpu['Model'])))
+        hw_lst.append(('cpu', "physical_{}".format(
+            processor), 'stepping', int(lscpu['Stepping'])))
+        for item in ['L1d cache', 'L1i cache', 'L2 cache', 'L3 cache']:
+            hw_lst.append(('cpu', "physical_{}".format(
+                processor), item.lower(), lscpu[item]))
+        hw_lst.append(('cpu', "physical_{}".format(
+            processor), 'min_Mhz', float(lscpu['CPU min MHz'])))
+        hw_lst.append(('cpu', "physical_{}".format(
+            processor), 'max_Mhz', float(lscpu['CPU max MHz'])))
+        hw_lst.append(('cpu', "physical_{}".format(
+            processor), 'current_Mhz', float(lscpu['CPU MHz'])))
+        hw_lst.append(('cpu', "physical_{}".format(
+            processor), 'flags', lscpu['Flags']))
+
+    hw_lst.append(('cpu', 'logical', 'number', int(lscpu['CPU(s)'])))
+    # Govenors could be differents on logical cpus
+    for cpu in range(int(lscpu['CPU(s)'])):
+        if scaling_governor_output is None:
+            scaling_governor = detect_utils.output_lines(
+                "cat /sys/devices/system/cpu/cpufreq/policy{}/scaling_governor".format(cpu))
+        else:
+            scaling_governor = scaling_governor_output
+        for line in scaling_governor:
+            hw_lst.append(('cpu', "logical_{}".format(
+                cpu), "governor", line.rstrip('\n')))
+
+    # Extracting numa nodes
+    hw_lst.append(('numa', 'nodes', 'count', int(lscpu['NUMA node(s)'])))
+    for numa in range(int(lscpu['NUMA node(s)'])):
+        cpus = lscpu['NUMA node{} CPU(s)'.format(numa)]
+        # lscpu -x provides the cpu mask
+        cpu_mask = lscpux['NUMA node{} CPU(s)'.format(numa)]
+        total_cpus = 0
+        # lscpu report numa nodes like 0-5,48-53
+        for item in cpus.split(','):
+            max = int(item.split("-")[1])
```

Comment:
Note that you're overriding the built-in functions `min` and `max` here.

Reply:
fixing this, thx.